core/serialize/serialize.cc

```diff
@@ -1108,6 +1109,7 @@ ast::TreePtr SerializerImpl::unpickleExpr(serialize::UnPickler &p, const GlobalS
             static_assert(sizeof(flags) == sizeof(flagsU1));
             // Can replace this with std::bit_cast in C++20
             memcpy(&flags, &flagsU1, sizeof(flags));
+            auto numPosArgs = static_cast<u2>(p.getU4());
```

Comment:
It's possibly worth enforcing that this fits in a u2 here, but the only situation that I think could cause that would be a corrupted serialized data.